bin/python_scripts/fix_organograms_s3_filenames.py

```diff
@@ -0,0 +1,264 @@
+        else:
+            cursor, _, _ = get_db_cursor(path)
+            if cursor.fetchone():
+                logger.info('Missing %s', new_path)
+
+        # deletion is run after s3 objs have already been copied and SOLR reindexed
+        # so skip other operations
+        return SKIP, NOT_PURGED
+
+    cursor, _, _ = get_db_cursor(path)
+    if cursor.fetchone():
+        if not dry_run:
+            if not exists_on_s3(bucket, new_path):
+                s3.Object(bucket.name, new_path).copy_from(ACL='public-read',
+                    CopySource='{}/{}'.format(bucket.name, path))
+                logger.info('S3 - copied in S3 from %s to %s', path, new_path)
+            else:
+                return DONT_SKIP, NOT_COPIED
+        else:
+            logger.info('S3 - to be copied in S3 from %s to %s', path, new_path)
+        return DONT_SKIP, COPIED
+    else:
+        return DONT_SKIP, NOT_COPIED
+
+
+def get_db_cursor(obj_key):
+    s3_url_prefix = os.environ.get('S3_URL_PREFIX')
+    original_path = obj_key
+
+    original_database_url = '{}{}'.format(s3_url_prefix, original_path)
+
+    cursor = connection.cursor()
+    cursor.execute("SELECT url FROM resource WHERE url='{}';".format(original_database_url))
+
+    return cursor, original_database_url, s3_url_prefix
+
+
+def update_database(obj, new_path, dry_run):
+    cursor, original_database_url, s3_url_prefix = get_db_cursor(obj.key)
+    new_database_url = '{}{}'.format(s3_url_prefix, new_path)
+
+    if cursor.fetchone():
+        sql_statement = "UPDATE resource SET url = '{}' WHERE url = '{}';".format(new_database_url,original_database_url)
+
+        if not dry_run:
+            cursor.execute('BEGIN TRANSACTION;')
+            cursor.execute(sql_statement)
+            cursor.execute('COMMIT TRANSACTION;')
+
+            cursor.execute("SELECT url FROM resource WHERE url='{}';".format(new_database_url))
+
+            if new_database_url in cursor.fetchone():
+                logger.info('DB - SUCCESS executing %s', sql_statement)
+                return (original_database_url, new_database_url)
+            else:
+                logger.info('DB - FAILED executing %s', sql_statement)
+        else:
+            logger.info('DB - DRY RUN: %s', sql_statement)
+            return (original_database_url, new_database_url)
+
+    elif exists_on_s3(obj.Bucket(), original_database_url):
+        logger.info('DB - ERROR Original url not found in table - %s', original_database_url)
+
+
+def reindex_solr(dataset_name, dry_run):
+    paster_command = 'paster --plugin=ckan search-index rebuild {} -c /etc/ckan/ckan.ini'.format(dataset_name)
+
+    logger.info('SOLR reindex - %s command: %s', 'Will run' if dry_run else 'Running', paster_command)
+
+    if not dry_run:
+        try:
+            subprocess.call(paster_command, shell=True)
+        except:
+            exception = sys.exc_info()[1]
+            logger.error('Subprocess Failed, exception occured: %s', str(exception))
```

Comment:
You can [pass the exception to the logger command like `error(..., exc_info=exception)`](https://docs.python.org/3/library/logging.html#logging.Logger.debug) and avoid having to do the interpolation yourself and it'll print a nice stacktrace.

Reply:
ok